src/server/DeviceServer.js

```diff
@@ -361,9 +362,16 @@ class DeviceServer {
       );
 
       const description = await device.getDescription();
-      const { uuid } = FirmwareManager.getAppModule(
-        description.systemInformation,
-      );
+      // const { uuid } = FirmwareManager.getAppModule(
+      //   description.systemInformation,
+      // );
+      try {
+        uuid = FirmwareManager.getAppModule(
+          description.systemInformation,
+        )[0];
```

Comment:
I know -- the code you added before is fine. The code you commented out should be here -- not the way you are using `getAppModule` now. It doesn't return an array and the code we had before was correct.  You can keep the old code in a try/catch to handle GCC.